### PR TITLE
Adds SymEngine::symbols() and allows Symbol.new to take a Ruby symbol

### LINKED ISSUE #33: Symbol.new should accept Ruby :symbols instead of just 'strings'
This would be more in keeping with Ruby convention.

e.g.,
```
x = Symbol.new(:x)
```

### PR DESCRIPTION
**Pull request by mohawkjohn**
Fixes #33 and adds a convenience function, `SymEngine::symbols` which allows us to create multiple symbols simultaneously.

### CODE REVIEW

File: spec/symbol_spec.rb [hunk 1]
```diff
@@ -1,9 +1,52 @@
...
+
+    context 'with an array of Ruby Strings as an argument' do
+      it 'returns an Enumerable of Symbol objects' do
+        @test_symbols_method.call(2, %w{x y})
+      end
+    end
+
+    context 'with a splatted argument' do
+      it 'returns an Enumerable of Symbol objects' do
+        @test_symbols_method.call(2, 'x', 'y')
```
Comment (certik) [inline spec/symbol_spec.rb]: I see that you test the following:
```
    #    x, y = SymEngine.symbols(%i{x y})
    #    x, y = SymEngine.symbols(%w{x y})
    #    x, y = SymEngine.symbols('x', 'y')
    #    x, y = SymEngine.symbols('x y')
```
Based on your comment above, should we also test these:
```
    #    x, y = SymEngine.symbols([:x, :y])
    #    x, y = SymEngine.symbols(['x', 'y'])
```
I can see that you test `SymEngine::Symbol.new('x')` and `SymEngine::Symbol.new(:x)` below.

**Comment by certik** [comment]
@mohawkjohn thanks for the PR! I left a minor comment. Otherwise it looks good.

File: spec/symbol_spec.rb (hunk 1 above)
Comment (mohawkjohn) [inline spec/symbol_spec.rb]: `%i{x y}` is the same as `[:x, :y]`, and `%w{x y}` is the same as `['x', 'y']`.

I suppose we could also be exhaustive and test `SymEngine.symbols(:x, :y)`, but it follows the same execution path as `SymEngine.symbols('x', 'y')`, so I think it's redundant.

Comment (certik) [inline spec/symbol_spec.rb]: Ah ok. Then all is good.

**Comment by certik** [comment]
There is an error in Ruby 1.9.3 (https://travis-ci.org/symengine/symengine.rb/jobs/121473492):
```
/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1361:in `load': /home/travis/gems/symengine-0.1.0/spec/symbol_spec.rb:33: unknown type of %string (SyntaxError)

        @test_symbols_method.call(2, %i{x y})

                                        ^

/home/travis/gems/symengine-0.1.0/spec/symbol_spec.rb:33: syntax error, unexpected $end, expecting ')'

        @test_symbols_method.call(2, %i{x y})
```
Is this syntax not supported in Ruby 1.9.3?

**Comment by abinashmeher999** [comment]
Thanks a lot @mohawkjohn! This makes the interface better.

**Comment by certik** [comment]
I checked that the first release of Ruby 1.9.3 was at the end of 2011. For example, the first release of Python 2.6 was in 2008, but we are dropping it in SymPy this year. Python 2.7 was first released in 2010, and we definitely still support it --- but Python 2.7 is special in that it is the last version of Python 2.x, so it will be around for a long, long time. @mohawkjohn, what is the situation in Ruby --- what is the earlier version that we should support?

**Comment by mohawkjohn** [comment]
Oh, bleh. I didn't realize `%i` was introduced in Ruby 2.0.

http://stackoverflow.com/questions/17355177/what-is-the-origin-of-i-notation

It's probably reasonable to support 1.9.3. We should definitely cut off before 1.9.1. I'd say just require 1.9.3 or higher for now.

I've fixed the commit and tested it with 1.9.3-p551 now.

**Comment by certik** [comment]
Thanks, that sounds reasonable. Let's support 1.9.3 and up. All tests pass, so I am merging this.

**Pull request merged**